algorithms/hhl/hhl/hhl.ipynb

```diff
@@ -1,982 +1,1292 @@
 {
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #2.    from binary_fractions import Binary
</pre><p>Is it possible to avoid the usage of the additional external package binary_fractions?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/194/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Reply:
<p>It is possible by writing specific function that will return binary representation of a fraction up to a given number of digits, e.g.: </p>

<pre class="ql-syntax" spellcheck="false">def fraction_to_binary(num, num_digits):
&nbsp; &nbsp; if num == 0:
&nbsp; &nbsp; &nbsp; &nbsp; return "0"
&nbsp; &nbsp; 
&nbsp; &nbsp; integer_part = int(num)
&nbsp; &nbsp; fractional_part = num - integer_part
&nbsp; &nbsp; 
&nbsp; &nbsp; binary_integer = bin(integer_part)[2:]
&nbsp; &nbsp; 
&nbsp; &nbsp; binary_fraction = []
&nbsp; &nbsp; if fractional_part &gt; 0:
&nbsp; &nbsp; &nbsp; &nbsp; binary_fraction.append(".")
&nbsp; &nbsp; &nbsp; &nbsp; while fractional_part and len(binary_fraction) &lt;= num_digits:
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; fractional_part *= 2
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; bit = int(fractional_part)
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; binary_fraction.append(str(bit))
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; fractional_part -= bit
&nbsp; &nbsp; 
&nbsp; &nbsp; return binary_integer + ''.join(binary_fraction)
</pre><p>Would you like me to add this function to the notebook or there will be similar function in the Classiq library soon?</p>

Reply:
<p>Yes, it is better to add the code. We might include something similar, but let us not wait for this.</p><p>Why can't you use <code>np.binary_repr</code>? something like <code>np.binary_repr(int(fraction*2**m))</code>  </p>

Reply:
Updated with use of `np.binary_repr`.